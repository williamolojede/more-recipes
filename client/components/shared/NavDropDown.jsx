import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavDropDown = ({ fullname, handleLogout }) => (
  <ul className="dropdown-menu z-depth-1">
    <li className="dropdown-header">Signed in as
      <p>{ fullname }</p>
    </li>
    <li className="dropdown-item divider" />
    <li className="dropdown-item"><Link to="/user">Your Profile</Link></li>
    <li className="dropdown-item"><Link to="/user/recipes">Your Recipes</Link></li>
    <li className="dropdown-item"><Link to="/user/favorites">Your Favorite Recipes</Link></li>
    <li className="dropdown-item divider" />
    <li className="dropdown-item"><Link to="/user/recipes/create">Create New Recipe</Link></li>
    <li className="dropdown-footer">
      <Link
        to="/logout"
        className="logout__button"
        onClick={handleLogout}
      >
        Sign Out
      </Link>
    </li>
  </ul>
);

NavDropDown.propTypes = {
  fullname: PropTypes.string.isRequired,
  handleLogout: PropTypes.func.isRequired
};

export default NavDropDown;
