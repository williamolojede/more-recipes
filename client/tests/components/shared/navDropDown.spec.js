import React from 'react';
import NavDropDown from '../../../components/shared/NavDropDown';
import userMock from '../../__mocks__/user.mock';

function setup() {
  const props = {
    fullname: userMock.fullname,
    handleLogout: jest.fn()
  };
  const wrapper = shallow(<NavDropDown {...props} />);

  return {
    wrapper
  };
}

describe('Nav Dropdown', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call a function to logout when the logout link is clicked', () => {
    const { wrapper } = setup();
    const logoutButton = wrapper.find('.logout__button');
    const handleLogoutSpy = jest.spyOn(wrapper.instance().props, 'handleLogout');
    expect(logoutButton.length).toBe(1);
    logoutButton.simulate('click');
    expect(handleLogoutSpy).toHaveBeenCalledTimes(1);
  });
});
