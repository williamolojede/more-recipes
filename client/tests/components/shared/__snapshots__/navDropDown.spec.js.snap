// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Dropdown should render with right amount of elements 1`] = `
<ul
  className="dropdown-menu z-depth-1"
>
  <li
    className="dropdown-header"
  >
    Signed in as
    <p>
      John Doe
    </p>
  </li>
  <li
    className="dropdown-item divider"
  />
  <li
    className="dropdown-item"
  >
    <Link
      replace={false}
      to="/user"
    >
      Your Profile
    </Link>
  </li>
  <li
    className="dropdown-item"
  >
    <Link
      replace={false}
      to="/user/recipes"
    >
      Your Recipes
    </Link>
  </li>
  <li
    className="dropdown-item"
  >
    <Link
      replace={false}
      to="/user/favorites"
    >
      Your Favorite Recipes
    </Link>
  </li>
  <li
    className="dropdown-item divider"
  />
  <li
    className="dropdown-item"
  >
    <Link
      replace={false}
      to="/user/recipes/create"
    >
      Create New Recipe
    </Link>
  </li>
  <li
    className="dropdown-footer"
  >
    <Link
      className="logout__button"
      onClick={[Function]}
      replace={false}
      to="/logout"
    >
      Sign Out
    </Link>
  </li>
</ul>
`;
